public/app/features/plugins/plugin_loader.ts

```diff
@@ -28,9 +28,13 @@ SystemJS.addImportMap({ imports });
 
 const systemJSPrototype: SystemJSWithLoaderHooks = SystemJS.constructor.prototype;
 
-// Monaco Editors reliance on RequireJS means we need to transform
-// the content of the plugin code at runtime which can only be done with fetch/eval.
-systemJSPrototype.shouldFetch = () => true;
+// This controls whether SystemJS will load a plugin with a script tag or fetch and eval.
+// We only want to fetch and eval plugins that are hosted on a CDN or are Angular plugins.
+systemJSPrototype.shouldFetch = function (url) {
+  const pluginInfo = getPluginFromCache(url);
+
+  return isHostedOnCDN(url) || Boolean(pluginInfo?.isAngular);
```

Comment:
is excluding CDN just for simplicity now or there is anything major that would prevent us from loading via script tags from CDN too?

Reply:
We cannot load CDN plugins via script tags right now because we transform the code at runtime to replace asset paths which is only possible with fetch and eval. It's gonna be an iterative process to get there.